lib/screens/home/accounts/accounts_screen.dart

```diff
@@ -25,31 +26,28 @@ class AccountsScreen extends StatefulWidget {
 class AccountsScreenState extends State<AccountsScreen> {
   final _listKey = GlobalKey<AnimatedListState>();
   final _subscriptions = CompositeSubscription();
-  final _accountsBloc = AccountsBloc();
   final _accountService = get<AccountService>();
+  late final AccountsBloc _bloc;
 
   @override
   void dispose() {
     _subscriptions.dispose();
-
     super.dispose();
-
-    get.unregister<AccountsBloc>();
   }
 
   @override
   void initState() {
     super.initState();
 
-    get.registerSingleton(_accountsBloc);
-
-    _accountsBloc.load();
-
     _accountService.events.removed.listen(_onRemoved).addTo(_subscriptions);
-    _accountsBloc.insert.listen(_onInsert).addTo(_subscriptions);
-    _accountsBloc.loading
-        .listen((e) => _onLoading(context, e))
-        .addTo(_subscriptions);
+  }
+
+  @override
+  void didChangeDependencies() {
+    _bloc = Provider.of<AccountsBloc>(context);
+    _bloc.insert.listen(_onInsert).addTo(_subscriptions);
```

Comment:
I think we will want to cancel existing subscriptions prior to adding more here. Same for other similar usages in `didChangeDependencies`.

Reply:
it should be ok. this is called when one of the items that the widget depends on changes. I think you are thinking of didUpdateWidget

Reply:
If we don't clear the old subscription, it seems like `_subscriptions` will still have a reference to the old bloc, creating a memory leak.

Reply:
Can we just have a second `CompositeSubscription` or use a `StreamSubscription`?

Reply:
Done.